CMakeLists.txt

```diff
@@ -493,6 +493,45 @@ endif()
 set(CPACK_PACKAGE_FILE_NAME "dfhack-${DFHACK_VERSION}-${DFHACK_BUILD_ID_PACKAGE}${DFHACK_PACKAGE_PLATFORM_NAME}-${DFHACK_BUILD_ARCH}${DFHACK_PACKAGE_SUFFIX}")
 INCLUDE(CPack)
 
+OPTION(DFHACK_INCLUDE_CORE "Download and include Dwarf Fortress core files in DFHack. Useful for local testing, but should not be used in releases." OFF)
+IF(DFHACK_INCLUDE_CORE)
+    STRING(REPLACE "." "_" DF_CORE_FILENAME "${DF_VERSION}")
+    STRING(REGEX REPLACE "^0_" "df_" DF_CORE_FILENAME "${DF_CORE_FILENAME}")
+    IF(UNIX)
+        IF(APPLE)
+            STRING(APPEND DF_CORE_FILENAME "_osx")
+        ELSE()
+            STRING(APPEND DF_CORE_FILENAME "_linux")
+        ENDIF()
+        IF(DFHACK_BUILD_32)
+            STRING(APPEND DF_CORE_FILENAME "32")
+        ENDIF()
+        STRING(APPEND DF_CORE_FILENAME ".tar.bz2")
+    ELSE()
+        STRING(APPEND DF_CORE_FILENAME "_win")
+        IF(DFHACK_BUILD_32)
+            STRING(APPEND DF_CORE_FILENAME "32")
+        ENDIF()
+        STRING(APPEND DF_CORE_FILENAME ".zip")
+    ENDIF()
+    SET(DF_CORE_URL "http://bay12games.com/dwarves/${DF_CORE_FILENAME}")
+    IF (NOT EXISTS "${CMAKE_BINARY_DIR}/${DF_CORE_FILENAME}")
+        FILE(DOWNLOAD "${DF_CORE_URL}" "${CMAKE_BINARY_DIR}/${DF_CORE_FILENAME}" SHOW_PROGRESS)
+    ENDIF()
+    FILE(REMOVE_RECURSE "${CMAKE_BINARY_DIR}/df-core")
```

Comment:
This will run whenever CMake runs - I assume this isn't intended for personal use, then? What is it intended for? As far as I know, DFHack only depends on a couple files in the DF folder at most, and they don't change often.

Reply:
The idea is that DFHack would install the correct version of DF automatically for easier development.

Is there a way it could extract the archive only if it's not already extracted from the same version?

Reply:
It would be difficult, and would probably rely on Toady updating the correct files. Or you could write your own file containing the version number and check that.
Still, clearing the DF folder would happen on new releases. It's not meant for people to play with, so what's it meant for?

Reply:
This isn't meant to be used with `make package`, but rather with `make install`. It removes the need for the DFHack dev to update their installation of Dwarf Fortress manually. For example, in Visual Studio, it's stored in `C:\Users\Owner\CMakeBuilds\f59d9b5c-4f13-7f38-a962-38d2a89cfda6\install\MSVC 64 Debug` for me.

Reply:
And why do you need to `make install` if you're not going to be using the resulting build? Does it help with packaging somehow?

Reply:
Oh, ok, so df-core is cleared sometimes but the installation directory isn't, and that's what's useful. Makes sense now, thanks.